### PR TITLE
WIP: components-first-release

### LINKED ISSUE #6: iterate components
We have transferred most of our components from the private repo to our new *mocking bird* project. Now, we need to finish the transition, bringing the remaining components to the project.

However, at the moment it is necessary to:

- [x] release the components already transferred

Next, it is necessary to transfer and implement the following components:

> @Felipe-gsilva, put a list here of the elements that you think are feasible to transfer during this sprint. Be concise and be careful not to give yourself more work than you think you can finish, based on your experience from the last sprint.

### PR DESCRIPTION
**Pull request by Felipe-gsilva**
fixes #6 

This pull request addresses issue #6 by finalizing the transition of components from the private repository to the new Mockingbird project. The key features and changes introduced in this PR include:

1. **Release Automation Workflows:**
   - **npm Release Workflow (`.github/workflows/npm.yml`):**
     - Added a GitHub Actions workflow to automate the process of publishing the Node.js package to the npm registry.
     - Triggers on new tags matching the pattern `v*` or manual dispatch.
     - Sets up the Node.js environment with version 20 and runs build commands to compile the library.
     - Utilizes `JS-DevTools/npm-publish@v3.1.1` action for publishing, securely using the `NPM_TOKEN` secret.
   - **Clojars Release Workflow (`.github/workflows/release.yml`):**
     - Added a workflow to automate the release process to Clojars for the Clojure project.
     - Includes steps to set up both Clojure and Node.js environments.
     - Executes build and deployment commands, using `CLOJARS_USERNAME` and `CLOJARS_PASSWORD` secrets for authentication.

2. **Inclusion of Target CSS Code:**
   - Integrated the target CSS code essential for component styling.
   - Updated the build process to ensure CSS assets are properly compiled and included in releases.
   - Added instructions in the `README.md` to copy the `tailwind.config.js` file for Tailwind CSS configuration.

3. **Documentation Enhancements (`README.md` and `docs/`):**
   - **`README.md` Updates:**
     - Expanded the installation section to cover both Clojars and npm distributions.
     - Provided detailed usage examples, including importing components and passing arguments.
     - Added links and notes to improve clarity on prerequisites and setup steps.
   - **Documentation Revisions:**
     - Removed outdated items from `docs/Roadmap.md` to reflect current project focus.
     - Updated `docs/test.md` to streamline local testing instructions, including command adjustments.

4. **Node.js Integration and Build Scripts:**
   - Set up Node.js environment using version 20 in the release workflows.
   - Included build scripts (`npm ci`, `npm run lib`, `npm run release`) to compile the library and prepare it for release.

5. **Security and Credentials Management:**
   - Utilized GitHub Secrets for managing sensitive information:
     - `NPM_TOKEN` for npm publishing.
     - `CLOJARS_USERNAME` and `CLOJARS_PASSWORD` for Clojars deployment.
   - Ensured that all credentials are securely handled within workflows.

6. **Cleanup and Refinements:**
   - Removed deprecated components and animations from the roadmap to align with the current project direction.
   - Adjusted commands and scripts to improve build consistency and reliability.

## Release Notes

- **New Features**
	- Introduced automated workflows for publishing Node.js and Clojure packages to npm and Clojars.
	- Added new components for rendering messages and profile pictures in the user interface.

- **Documentation**
	- Enhanced clarity and usability of the README with installation instructions and usage examples.
	- Updated the roadmap and test documentation for better guidance.

- **Bug Fixes**
	- Improved responsiveness and layout adjustments in the hero component.

- **Refactor**
	- Renamed and exported several components for better accessibility and usability.

### CODE REVIEW

File: src/mockingbird/node.cljs [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+(defn hello []
+  (prn "hello")
+  "hello")
+
+(defn generate-exports []
+  #js{:hello hello
+      :helix-dom d/img
+      :button button
+      :image pfp})
```
[the remark this answers is not in the document]
Comment (avelino) [reply src/mockingbird/node.cljs:15]: `pfp` it's not a good name, I agree with the review

> explicit is better than implicit

File: .github/workflows/npm.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+jobs:
+    publish:
+        runs-on: ubuntu-latest
+        steps:
+            - uses: actions/checkout@v4
+            - uses: actions/setup-node@v3
+            - uses: "DeLaGuardo/setup-clojure@12.5"
+              with:
+                node-version: "20"
```
[the remark this answers is not in the document]
Comment (Felipe-gsilva) [reply .github/workflows/npm.yml:17]: thank you bot, that code is really inverted

File: build.clj [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+(defn jar [_]
+  (b/write-pom {:class-dir class-dir
+                :lib lib
+                :version version
+                :basis basis
+                :src-dirs ["src"]})
+  (b/copy-dir {:src-dirs ["src" "resources"]
+               :target-dir class-dir})
+  (b/jar {:class-dir class-dir
+          :jar-file jar-file}))
```
[the remark this answers is not in the document]
Comment (Felipe-gsilva) [reply build.clj:23]: they are not part of the build

File: webpack.config.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
+const path = require('path');
+
+module.exports = (env) => {
```
[the remark this answers is not in the document]
Comment (avelino) [reply webpack.config.js:3]: @Felipe-gsilva apparently this suggestion makes sense

File: README.md [hunk 5]
```diff
@@ -16,38 +18,71 @@ It's impossible to talk about *"mockingbird"* without remembering the song [Mock
...
-```
 
-- configure your shadow cljs:
+$ npx create-cljs-project your-project
+$ npm install react autoprefixer babel-loader css-loader cssnano karma karma-chrome-launcher karma-cljs-test msw npm-run-all postcss postcss-cli postcss-loade postcss-preset-env shadow-cljs tailwindcss webpack webpack-cli rimraf glob @isaacs/cliui --save-dev
 
-> for better understanding, follow the offcial [guide](https://github.com/thheller/shadow-cljs)
+```
 
+Copy our [tailwind.config.js](tailwind.config.js), specially the colors, into your own config.
```
[the remark this answers is not in the document]
Comment (Felipe-gsilva) [reply README.md:52]: this suggestion block is broken @coderabbitai

File: docs/test.md [hunk 6]
```diff
@@ -1,8 +1,8 @@
 # Running the local mocking bird tests
 
-Mockingbird is a powerful tool for helping you build your project, but, as a great dev, you will not insert any random tool someone told you to. Instead, you are probably gonna test it several times and also contribute to the open source code, won't you?
+Mockingbird is a powerful tool for helping you build your project, but, as a great dev, you will not insert any random tool someone told you to. Instead, you will test it several times and also contribute to the open source code, won't you?
```
[the remark this answers is not in the document]
Comment (Felipe-gsilva) [reply docs/test.md:3]: there is no need to do this, @coderabbitai

**Comment by Felipe-gsilva** [comment]
@coderabbitai, can you change the PR description adding more context around some of the important features? As I can see, the target css code is not being mentioned neither some other code changes

**Comment by Felipe-gsilva** [comment]
# To Do
## 0.0.1 launching

- npm compile fix -> webpack bundle is being generated, yet, there are some minor issues while importing the code on another react/shadow-cljs+helix project
- target.css -> need to check if any difference has occurred in the now used file and the one from source at compile time, since for now, the css code generated from mockingbird is only copied to user project once (or when this file deleted)
- tailwind.config.js, postcss and these other config for the users. Now, people have to manually gather the config files from our codebase and it seems kind of a frustation to copy things that do not look at first necessary

## quality of life improvements

- another themes for mockingbird -> for now, we only have the default mockingbird theme, which is great, but it would be nicier to have the other 2 themes as intended before.|
- loading? state being updated at all components -> for now, even though some components can have its states changed during runtime, it do not mostly affects UI/UX
- style all the components
- aside fix

## for future
- make the code reusable in all React projects.
- make example page better, as a playground for new components

**Review by J0sueTM — APPROVED** [review]
LGTM!

thank god we setted the bot on chill mode, it has gotten really hard lately to actually review the code with all it's comments on the way.

**Comment by J0sueTM** [comment]
i wrongfully wrote God with a minute g. Bad moments coming up :disappointed:

**Pull request merged**